src/services/downloadManager.ts

```diff
@@ -65,20 +68,36 @@ class DownloadManager {
                         { responseType: 'arraybuffer' }
                     );
                     const worker = await new CryptoWorker();
-                    const decrypted: any = await worker.decryptFile(
+                    const decryptedFile: any = await worker.decryptFile(
                         new Uint8Array(resp.data),
                         await worker.fromB64(file.file.decryptionHeader),
                         file.key
                     );
-                    return URL.createObjectURL(new Blob([decrypted]));
+                    let decryptedFileBlob = new Blob([decryptedFile]);
+
+                    if (getFileExtension(file.metadata.title) === TYPE_HEIC) {
+                        decryptedFileBlob = await this.convertHEIC2JPEG(
+                            decryptedFileBlob
+                        );
+                    }
+                    return URL.createObjectURL(decryptedFileBlob);
                 } catch (e) {
-                    console.log('get file failed ' , e );
+                    console.log('get file failed ', e);
                 }
             })();
             this.fileDownloads.set(file.id, download);
         }
         return await this.fileDownloads.get(file.id);
     };
+
+    private async convertHEIC2JPEG(fileBlob): Promise<Blob> {
+        const heic2any = require('heic2any');
```

Comment:
We could define this at the top since DownloadManager already maintains state? Will prevent reinitializations.

Reply:
There was issue where the next build failed when i tried to load this library. The problem was that the library used some browser defined things(Blob) so, the pre-render on server side failed. 
Found a way to skip that (edit pushed in the next commit)